src/components/CarouselItem.js

```diff
@@ -12,11 +13,50 @@ class CarouselItem extends PureComponent {
     width: PropTypes.number,
     offset: PropTypes.number,
     index: PropTypes.number,
+    id: PropTypes.number,
     currentSlideIndex: PropTypes.number,
     isDragging: PropTypes.bool,
     isDraggingEnabled: PropTypes.bool,
   };
 
+  constructor(props) {
+    super(props);
+    this.childrenRef = React.createRef();
+  }
+
+  /* ========== Resize, if necessary. Workaround for iOS safari ========== */
+  componentDidMount() {
+    this.observeWidth();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.width !== this.props.width) {
+      this.resizeChildren();
+    }
+  }
```

Comment:
Does this method do the same as `observeWidth`? Do we need it?

Reply:
`observeWidth` just observes the component for a short time period (from mounting until complete loading of the images). Then the observer is stopped (no battery drain on iPhone 5 as it doesn't support the ResizeObserver natively). When the view port is resized (e.g. turn the phone around) this method ensures, that the size is updated.